### PR TITLE
Bump EPEL dependency to '< 5' for EL9 support

### PR DESCRIPTION
**Pull request by jovandeginste**
Signed-off-by: Jo Vandeginste <Jo.Vandeginste@kuleuven.be>

### CODE REVIEW

File: metadata.json
Comment (kajinamit) [inline metadata.json:35]: Can we keep the original multi-line format ? Main concern is that this is touching unrelated lines because of the format change.

**Review by kajinamit — CHANGES_REQUESTED** [review]

**Comment by kajinamit** [comment]
It seems collectd is not yet released in EPEL 9 and that's why the CentOS 9 tests are failing.

File: metadata.json
Comment (jovandeginste) [reply metadata.json:35]: Yes - my auto-formatter does this. I added a "plugin" that keeps arrays multiline now.

**Comment by jovandeginste** [comment]
> It seems collectd is not yet released in EPEL 9 and that's why the CentOS 9 tests are failing.

So we wait until it lands, one way or another...

File: metadata.json
Comment (kajinamit) [reply metadata.json:35]: thanks. looks better now !

File: metadata.json
Comment (bastelfreak) [inline metadata.json:25]: can you please move this commit to a new PR? allowing newer epel isn't related to implementing EL9 support.

Comment (jovandeginste) [reply metadata.json:25]: > can you please move this commit to a new PR? allowing newer epel isn't related to implementing EL9 support.

puppet-epel only supports EL9 sinds [v4.1.0](https://github.com/voxpupuli/puppet-epel/blob/master/CHANGELOG.md#v410-2022-03-26); I feel this PR is senseless without updating this...

Comment (jovandeginste) [reply metadata.json:25]: Also, it (bump epel) is the very title of this PR... I must be too confused by the heat and it being a Friday afternoon.

Comment (traylenator) [reply metadata.json:25]: If 4.1.0 is minumum for EPEL9 then the lower bound should be increased also accordingly.

As for if this it should be in this MR at all I think its okay.

**Comment by traylenator** [comment]
collectd is now in EPEL 9 it seems so this is good to go once comments are addressed.

File: metadata.json
Comment (jovandeginste) [reply metadata.json:25]: > If 4.1.0 is minumum for EPEL9 then the lower bound should be increased also accordingly.

I understand the idea, but I'm hesitant here - you are not forced to use EL9, so maybe you want to keep using your current module versions. I know what a hell it is to try and update your modules when they all have some variation of requirements of a shared module (like epel).

Anyway, do you want me to increment the lower bound?

Comment (bastelfreak) [reply metadata.json:25]: this PR contains two changes. Adding EL 9 and allowing the latest EPEL module. Allowing  the latest EPEL module is a requirement for adding EL 9 support. But the change is important on it's own. the collectd module needs to allow the latest EPEL version in case it gets installed into an environment where other modules already require EPEL 4. because of that we should split this PR up into two. one allowing EPEL 4, the other adding code for EL 9.

**Comment by jovandeginste** [comment]
Create two new PRs to split this one:

- https://github.com/voxpupuli/puppet-collectd/pull/1004
- https://github.com/voxpupuli/puppet-collectd/pull/1005

**Pull request closed without merging**
